vscode-lean4/media/manual.md

```diff
@@ -0,0 +1,631 @@
+
+| &nbsp; |
+| :--: | 
+| ![](media/command-palette.png) | 
+| *VS Code command palette* |
+
+### Chords
+
+Many commands in VS Code are bound to chord keyboard shortcuts where multiple keys must be pressed in succession to trigger the command. Chords in VS Code are typically of the form `Ctrl+K Ctrl+<some other key>` (`Cmd+K Cmd+<some other key>`), which can be triggered by holding `Ctrl` (`Cmd`), pressing `K`, releasing `K` and then pressing `<some other key>`.
+
+### Configuring keyboard shortcuts
+
+The [keyboard shortcuts](command:workbench.action.openGlobalKeybindings) for all commands can be configured by navigating to 'File' > 'Preferences' > 'Keyboard Shortcuts' or by using the `Ctrl+K Ctrl+S` (`Cmd+K Cmd+S`) [chord](#chords). Typing in 'Lean 4' will display all commands of the Lean 4 VS Code extension.
+
+| &nbsp; |
+| :--: | 
+| ![](media/shortcut-settings.png) | 
+| *VS Code keyboard shortcut settings* |
+
+---
+
+## Interacting with Lean files
+
+This section describes how to use the Lean 4 VS Code extension to interact with, read and navigate individual Lean files.
+
+### File processing
+
+When opening a Lean file, Lean must first process the file in order to provide most interactive features for it. Since Lean files can contain arbitrary automation that needs to be executed when processing the file, doing so can be potentially expensive. 
+
+The current progress for processing the file is displayed using an orange bar on the right side of the editor as part of the scroll bar. Areas of the file that are marked with the orange bar are still being processed, whereas areas of the file where the orange bar has already disappeared have finished processing. An orange bar on the left side of the editor displays which of the source code lines that are currently visible in the editor are still being processed.
+
+When making a change to a file, Lean needs to re-process all declarations that can depend on the declaration that the change was made to. Currently, this means that all code below the point where the change occured needs to be re-processed.
+
+For parts of the file that are currently still being processed, Lean cannot provide any interactive features that require file processing information yet. For example, in unprocessed parts of the file, [errors, warnings and information](#errors-warnings-and-information) are not displayed, the [InfoView](#infoview) will not show the current goal state and [hovers](#hovers) will not display a popup until file processing has progressed to the respective part of the file.
+
+| &nbsp; |
+| :--: | 
+| ![](media/file-progress.png) | 
+| *File processing progress. The orange bar on the right displays the progress for the whole file, whereas the one on the left displays the progress for the currently visible source code lines.* |
+
+### Errors, warnings and information
```

Comment:
The [Error Lens](https://marketplace.visualstudio.com/items?itemName=usernamehw.errorlens) extension is quite popular with Lean users. Maybe it makes sense to mention it in this section?

Reply:
I've tested it out for a couple of months, but the main drawback I found was that most Lean errors are typically formatted to span multiple lines, so Error Lens is rarely helpful to actually understand the issue. I can still add it, though.